### PR TITLE
Add Rest API Structure for User Authentication 

### PR DESCRIPTION
**Pull request by ankit2001**
This PR aims to add all basic API structure of registration, login , logout, user profile, and Social Apps.

Please review it @fredfalcon , @souravbadami .

Thanks

### CODE REVIEW

**Review by souravbadami — CHANGES_REQUESTED** [review]
Lint tests are failing, could you please fix them?

**Comment by ankit2001** [comment]
Yeah, I will fix them and add other APIs also.



[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality5&>
16/06/21,
03:52:17 pm

File: bugheist/settings.py [hunk 1]
```diff
@@ -272,7 +272,7 @@
 SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
 
 # Allow all host headers
-ALLOWED_HOSTS = [".bugheist.com", "127.0.0.1"]
+ALLOWED_HOSTS = ["*"]
```
Comment (fredfalcon) [inline bugheist/settings.py:275]: Please add this back, it makes it more secure

File: bugheist/settings.py [hunk 2]
```diff
@@ -422,3 +433,12 @@
 STRIPE_LIVE_MODE = False  # Change to True in production
 
 DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
+
+CALLBACK_URL_FOR_GITHUB = os.environ.get(
+    "CALLBACK_URL_FOR_GITHUB", default="http://127.0.0.1:8000")
```
Comment (fredfalcon) [inline bugheist/settings.py:438]: What are these supposed to be?

**Review by fredfalcon — CHANGES_REQUESTED** [review]

File: bugheist/settings.py (hunk 2 above)
Comment (ankit2001) [reply bugheist/settings.py:438]: There are the callback urls for github and google, after authentication through google or github, they will redirect a user to some page with a code, we can use that code for access token, also can update that url to any thing where we want user to redirect after authentication.

File: bugheist/settings.py (hunk 1 above)
Comment (ankit2001) [reply bugheist/settings.py]: I have done with this @fredfalcon .

File: bugheist/settings.py (hunk 2 above)
Comment (fredfalcon) [reply bugheist/settings.py:438]: Can you default them to the base url?

Comment (ankit2001) [reply bugheist/settings.py:438]: Yes, we can remail them as it is.

Comment (ankit2001) [reply bugheist/settings.py:438]: Yes we can remail them localhost or default

Comment (ankit2001) [reply bugheist/settings.py:438]: Yes we can remain them localhost or default

**Comment by fredfalcon** [comment]
Please fix the conflicts and it’s good to go, thanks!

**Review by fredfalcon — APPROVED** [review]

**Review by souravbadami — APPROVED** [review]

**Pull request merged**